addons/mail/static/src/models/thread.js

```diff
@@ -1958,6 +1965,13 @@ registerModel({
         isDescriptionEditableByCurrentUser: attr({
             compute: '_computeIsDescriptionEditableByCurrentUser',
         }),
+        /**
+         * States whether current partner has already joined the channel or not.
+         */
+        isJoined: attr({
+            compute: '_computeHasCurrentParternerJoinedChannel',
+            default: false,
+        }),
```

Comment:
I believe you should just use the "pinned" state in your condition and not introduce this extra field.

Reply:
Indeed you were right. `isPinned` works fine.